nuxeo-drive-core/src/main/java/org/nuxeo/drive/listener/NuxeoDriveGroupUpdateListener.java

```diff
@@ -0,0 +1,128 @@
+import org.nuxeo.ecm.core.event.PostCommitFilteringEventListener;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.ecm.platform.usermanager.UserManager;
+import org.nuxeo.ecm.platform.usermanager.UserManagerImpl;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Post-commit asynchronous listener that handles group change events fired by the {@link UserManager}.
+ * <p>
+ * For all the documents carrying an ACL impacted by a changed group or one of its ancestors it fires the
+ * {@link NuxeoDriveEvents#GROUP_UPDATED} event that is handled by the synchronous
+ * {@link NuxeoDriveFileSystemDeletionListener}.
+ *
+ * @since 9.2
+ */
+public class NuxeoDriveGroupUpdateListener implements PostCommitFilteringEventListener {
+
+    protected List<String> acceptedEvents = Arrays.asList(UserManagerImpl.GROUPCREATED_EVENT_ID,
+            UserManagerImpl.GROUPDELETED_EVENT_ID, UserManagerImpl.GROUPMODIFIED_EVENT_ID);
+
+    @Override
+    public boolean acceptEvent(Event event) {
+        return event.getContext() != null
+                & UserManagerImpl.USER_GROUP_CATEGORY.equals(
+                        event.getContext().getProperty(DocumentEventContext.CATEGORY_PROPERTY_KEY))
+                && acceptedEvents.contains(event.getName());
+    }
+
+    @Override
+    public void handleEvent(EventBundle events) {
+        for (Event event : events) {
+            EventContext context = event.getContext();
+            if (context == null) {
+                return;
+            }
+            String groupName = (String) context.getProperty(UserManagerImpl.ID_PROPERTY_KEY);
+            if (StringUtils.isBlank(groupName)) {
+                return;
+            }
+            List<String> groupNames = getAllGroupNames(groupName, context);
+            handleUpdatedGroups(groupNames);
+        }
+    }
+
+    /**
+     * Returns a list containing the names of the given group and all its ancestor groups.
+     */
+    @SuppressWarnings("unchecked")
+    protected List<String> getAllGroupNames(String groupName, EventContext context) {
+        List<String> groupNames = new ArrayList<String>();
+        groupNames.add(groupName);
+        // Get ancestor groups from the event context or compute them if not provided
+        List<String> ancestorGroups = (List<String>) context.getProperty(UserManagerImpl.ANCESTOR_GROUPS_PROPERTY_KEY);
+        if (ancestorGroups != null) {
+            groupNames.addAll(ancestorGroups);
+        } else {
+            groupNames.addAll(Framework.getService(UserManager.class).getAncestorGroups(groupName));
+        }
+        return groupNames;
+    }
+
+    protected void handleUpdatedGroups(List<String> groupNames) {
+        RepositoryManager repositoryManager = Framework.getService(RepositoryManager.class);
+        for (String repositoryName : repositoryManager.getRepositoryNames()) {
+            CoreSession session = CoreInstance.openCoreSessionSystem(repositoryName);
+            try {
```

Comment:
Because I just applied https://github.com/nuxeo/nuxeo/blob/master/nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/CoreInstance.java#L91 but you're right, and I'll use `CoreInstance.doPrivileged` anyway.